agenda-services/src/main/java/org/exoplatform/agenda/notification/builder/AgendaTemplateBuilder.java

```diff
@@ -139,43 +141,43 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
-    icsContent += "DTEND:" + dateFormat.format(endDate) + "\n";
-    String description = "";
-    if (notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_DESCRIPTION) != null) {
-      description = notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_DESCRIPTION).replace("\n", "\\n");
-    }
-    if (notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_CONFERENCE) != null) {
-      if (!description.isEmpty()) {
-        description += "\\n";
-      }
-      description += "Webconference : " + notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_CONFERENCE);
-    }
-    if (!description.isEmpty()) {
-      icsContent += "DESCRIPTION:" + description + "\n";
-    }
-    if (notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_LOCATION) != null) {
-      icsContent += "LOCATION:" + notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_LOCATION) + "\n";
+    /* Generate unique identifier */
+    UidGenerator ug = new RandomUidGenerator();
+    Uid uid = ug.generateUid();
+    /* Create the event */
+    String eventSummary = notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_TITLE);
+    String startDateRFC3339 = notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_START_DATE);
+    String endDateRFC3339 = notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_END_DATE);
+    ZonedDateTime startDate = ZonedDateTime.parse(startDateRFC3339).withZoneSameInstant(timeZone);
+    ZonedDateTime endDate = ZonedDateTime.parse(endDateRFC3339).withZoneSameInstant(timeZone);
+    TimeZone ical4jTimezone = getICalTimeZone(timeZone);
+    DateTime startDateTime = new DateTime(Date.from(startDate.toInstant()),ical4jTimezone);
+    DateTime endDateTime = new DateTime(Date.from(endDate.toInstant()),ical4jTimezone);
+    VEvent vEvent = new VEvent(startDateTime, endDateTime, eventSummary);
+    vEvent.getProperties().add(uid);
+    /* Create calendar */
+    Calendar calendar = new Calendar();
+    calendar.getProperties().add(new ProdId("PRODID:-//eXo Plaform//EN"));
+    calendar.getProperties().add(Version.VERSION_2_0);
+    calendar.getProperties().add(CalScale.GREGORIAN);
+    Organizer organizer = new Organizer(URI.create("noreply@noreply.com"));
+    organizer.getParameters().add(new Cn(notification.getValueOwnerParameter(STORED_PARAMETER_EVENT_CREATOR)));
+    calendar.getProperties().add(organizer);
+    /* Add event to calendar */
+    calendar.getComponents().add(vEvent);
+    CalendarOutputter outputter = new CalendarOutputter();
+    ByteArrayOutputStream output =  new ByteArrayOutputStream();
+    try {
+      outputter.output(calendar, output);
+      byte[] bytes = output.toByteArray();
+      attachment.setInputStream(new ByteArrayInputStream(bytes));
+      attachment.setMimeType("text/calendar;charset=utf-8;method=PUBLISH");
+      messageInfo.addAttachment(attachment);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
```suggestion
      throw new IllegalStateException("Unable to convert event '" + eventSummary + "' to iCal format", e);
```
When throwing an exception, you have to add a useful information, in addition to the original exception. BTW, please avoid throwing Exception or RuntimeException since it's too generic as exceptions.